olp-cpp-sdk-core/include/olp/core/utils/Dir.h

```diff
@@ -124,6 +124,17 @@ class CORE_API Dir {
    * @return The calculated size.
    */
   static uint64_t Size(const std::string& path, FilterFunction filter_fn = {});
+
+  /**
+   * @brief Checks if the current application and user has a read only access to
+   * given path.
```

Comment:
If this expects that the path is a file and it only works on files then you need to state this here. Preferably you support both directories & files.

Reply:
This works for directories too.